modules/apps/translation/translation-service/src/main/java/com/liferay/translation/internal/model/listener/TranslationLayoutModelListener.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.translation.internal.model.listener;
+
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.translation.service.TranslationEntryLocalService;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Adolfo Pérez
+ */
+@Component(service = ModelListener.class)
+public class TranslationLayoutModelListener extends BaseModelListener<Layout> {
```

Comment:
Question:

I feel the  name _TranslationLayoutModelListener_ odd. I my opinion must be _LayoutModelListener_

But if you think that it's ok, I see the code OK

Reply:
We already have a TranslationJournalArticleModelListener in that same package, so I used the same name for consistency.

Reply:
![](https://media3.giphy.com/media/tsgNNs93oIbwk/giphy.gif)